### PR TITLE
feat: add suggestions to module not found error messages for file urls

### PR DESCRIPTION
**Pull request by dsherret**
For example:

```shellsession
> cat example.ts
import { Example } from "./routes";
console.log(Example);
> ../deno/target/debug/deno run example.ts                          
error: Module not found "file:///V:/scratch/routes". Maybe add a '.ts' extension or run with --unstable-sloppy-imports
    at file:///V:/scratch/example.ts:1:25
> ../deno/target/debug/deno run --unstable-sloppy-imports example.ts
Warning Sloppy import resolution (hint: add .ts extension)
    at file:///V:/scratch/example.ts:1:25
[class Example]
```

### CODE REVIEW

**Comment by ry** [comment]
are there any performance impacts? startup time?

**Comment by dsherret** [comment]
This code is only executed on error.

File: cli/graph_util.rs [hunk 1]
```diff
@@ -897,4 +972,46 @@ mod test {
...
+    assert_eq!(
+      sloppy_import_resolution_to_suggestion_message(
+        &SloppyImportsResolution::JsToTs(
+          ModuleSpecifier::parse("file:///dir/index.mts").unwrap()
+        )
+      )
+      .unwrap(),
+      "Maybe change the extension to '.mts'"
+    );
+  }
```
Comment (bartlomieju) [inline cli/graph_util.rs:1016]: Great 👍

File: cli/tests/integration/lsp_tests.rs [hunk 2]
```diff
@@ -10668,3 +10668,55 @@ fn lsp_sloppy_imports_warn() {
...
+          }
+        },
+        severity: Some(lsp::DiagnosticSeverity::ERROR),
+        code: Some(lsp::NumberOrString::String("no-local".to_string())),
+        source: Some("deno".to_string()),
+        message: format!(
+          "Unable to load a local module: {}\nMaybe add a '.ts' extension.",
+          temp_dir.join("a").uri_file(),
+        ),
+        ..Default::default()
```
Comment (bartlomieju) [inline cli/tests/integration/lsp_tests.rs:10716]: Oh, this is super nice 👍 will it still provide a quick-fix?

Comment (dsherret) [reply cli/tests/integration/lsp_tests.rs:10716]: Not at the moment. I will do that in a follow-up sometime.

**Comment by bartlomieju** [comment]
Right, but what if you fix all these errors and still run with the flag. Won't that cause unnecessary computation?

**Comment by dsherret** [comment]
> Right, but what if you fix all these errors and still run with the flag. Won't that cause unnecessary computation?

Yeah, good point. I'll do another PR that adds this soon.

**Pull request merged**